include/toml++/impl/path.h

```diff
@@ -23,20 +23,59 @@ TOML_NAMESPACE_START
 	using path_component_value = std::variant<std::string, size_t>;
 
 	/// \brief Represents a single component of a complete 'TOML-path': either a key or an array index
-	struct path_component
+	struct TOML_EXPORTED_CLASS path_component
 	{
-		path_component_value value;
-		path_component_type type;
+		friend class path;
 
 	  private:
 		/// \cond
 
+	  	path_component_value value_;
+		path_component_type type_;
+
 		TOML_PURE_GETTER
 		TOML_EXPORTED_STATIC_FUNCTION
 		static bool TOML_CALLCONV equal(const path_component&, const path_component&) noexcept;
 
 		/// \endcond
 	  public:
+		/// \brief	Default constructor.
+		TOML_NODISCARD_CTOR
+		path_component() noexcept = default;
+
+		/// \brief	Constructor for a path component that is an array index
+		TOML_NODISCARD_CTOR
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path_component(size_t index);
+
+		/// \brief	Constructor for a path component that is a key string
+		TOML_NODISCARD_CTOR
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path_component(std::string_view key);
+
+#if TOML_ENABLE_WINDOWS_COMPAT
+
+		/// \brief	Constructor for a path component that is a key specified witha wide-char string
+		TOML_NODISCARD_CTOR
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path_component(std::wstring_view key);
+
+#endif
+
+		/// \brief Retrieve the value of this path component, either the key name or the aray index
+		TOML_PURE_INLINE_GETTER
+		const path_component_value& get_value() const noexcept
+		{
+			return value_;
+		}
+
+		/// \brief Retrieve the type of this path component, either path_component::key or path_component::array_index
+		TOML_PURE_INLINE_GETTER
+		path_component_type get_type() const noexcept
+		{
+			return type_;
+		}
```

Comment:
```suggestion
		/// \brief Retrieve the value of this path component, either the key name or the aray index
		TOML_PURE_INLINE_GETTER
		const path_component_value& value() const noexcept
		{
			return value_;
		}

		/// \brief Retrieve the type of this path component, either path_component::key or path_component::array_index
		TOML_PURE_INLINE_GETTER
		path_component_type type() const noexcept
		{
			return type_;
		}
```

I generally avoid the `get_XXX()` pattern in the public API - functions that take no arguments and simply return a value are inherently getters by their very nature; having "get_" in their name doesnlt really add any value

(there are some exceptions, but those are mostly just called "get()" and make sense in context)